pkg/controller/job/job_controller_test.go

```diff
@@ -5172,6 +5172,66 @@ func TestFinalizersRemovedExpectations(t *testing.T) {
 	}
 }
 
+func TestBackupFinalizerRemoval(t *testing.T) {
+	_, ctx := ktesting.NewTestContext(t)
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	clientset := fake.NewSimpleClientset()
+	sharedInformers := informers.NewSharedInformerFactory(clientset, controller.NoResyncPeriodFunc())
+	manager := NewController(ctx, sharedInformers.Core().V1().Pods(), sharedInformers.Batch().V1().Jobs(), clientset)
+	manager.podStoreSynced = alwaysReady
+	manager.jobStoreSynced = alwaysReady
+
+	go manager.Run(context.TODO(), 0)
```

Comment:
```suggestion
	go manager.Run(ctx, 0)
```

Reply:
Also add a comment to indicate why the number of workers is `0` as this seems quite unique. Maybe something like: `// initialize the controller with 0 workers to make sure the pod finalizers are not removed by the "syncJob" function`.